### PR TITLE
First POC for refactored hive cookbook

### PR DESCRIPTION
**Pull request by ekund**
This is a PR for comment purposes to discuss whether our new strategy for refactoring makes sense.

### CODE REVIEW

File: cookbooks/bach_hive/metadata.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name 'bach_hive'
+maintainer 'BloombergLP'
+maintainer_email 'eiserovich1@bloomberg.net'
```
Comment (amithkanand) [inline cookbooks/bach_hive/metadata.rb]: this should be `hadoop@bloomberg.net`

File: cookbooks/bach_hive/providers/alternatives.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require 'mixlib/shellout'
+
+use_inline_resources
+
+action :create do
+  install = Mixlib::ShellOut.new('update-alternatives --install' \
+      " /etc/#{new_resource.component}/conf " \
+      " #{new_resource.component}-conf " \
+      " /etc/#{new_resource.component}/conf.#{new_resource.link_name} 50")
+  install.run_command
```
Comment (amithkanand) [inline cookbooks/bach_hive/providers/alternatives.rb]: Do we need to check `success` or `failure` of the executed command?

File: cookbooks/bach_hive/providers/alternatives.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+  set = Mixlib::ShellOut.new('update-alternatives --set ' \
+      " #{new_resource.component}-conf " \
+      "/etc/#{new_resource.component}/conf.#{new_resource.link_name}")
+  set.run_command
```
Comment (amithkanand) [inline cookbooks/bach_hive/providers/alternatives.rb]: Do we need to check `success` or `failure` of the executed command?

File: cookbooks/bach_hive/recipes/config.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+%w(hive webhcat hcat hive-hcatalog).each do |w|
+  directory "/etc/#{w}/conf.#{node.chef_environment}" do
+    owner 'root'
+    group 'root'
+    mode 00755
+    action :create
+    recursive true
+  end
+
+  bach_hive_alternatives "update-#{w}-conf-alternatives" do
```
Comment (amithkanand) [inline cookbooks/bach_hive/recipes/config.rb]: Should we move alternatives to `Library` cookbook? This will be used by multiple cookbooks/components.

**Comment by http-418** [comment]
This is great.  This is exactly the kind of chef code I like to see.  It's clear, it's based around re-usable providers, and it comes with tests.

That said
- Did you evaluate the community 'alternatives' cookbook before you wrote your own provider?

- I'd really rather not start chipping away at the rubocop rules.  In particular, hashrocket syntax is deprecated.

- Did you consider https://github.com/poise/poise instead of using LWRPs?

**Comment by vt0r** [comment]
Closing for cleanup. Feel free to reopen if this work gets picked up later on.

**Pull request closed without merging**